lib/components/MiningDashboard.js

```diff
@@ -33,20 +33,16 @@ class MiningDashboard extends React.PureComponent {
   }
 
   async componentDidMount() {
+    this.client = getClient().node;
     const mining = await this.client.execute('getgenerate');
     const miningInfo = await this.client.execute('getmininginfo');
     this.setState({ mining, miningInfo });
   }
 
-  onUpdateCoinbase(e) {
-    e.preventDefault();
-    this.setState({ coinbaseAddress: e.target.value });
-  }
-
-  async onSetAddress(e) {
-    const { coinbaseAddress } = this.state;
-    window.sessionStorage.setItem('coinbaseAddress', coinbaseAddress);
-    this.setState({ savedCoinbase: coinbaseAddress });
+  async onSetAddress() {
+    const { address } = this.state;
+    this.props.updateCoinbaseAddress(address);
+    this.setState({ address: '' });
```

Comment:
Why setting address to empty here?

Reply:
It clears the input area once the coinbase address has been set. makes for a smoother user experience I feel like.

Reply:
I think adding a comment saying that for our future selves would be really useful